tests/collecting/ImportersTest.py

```diff
@@ -2,25 +2,65 @@
 import unittest
 from collections import OrderedDict
 
+from importlib.machinery import (
+    ModuleSpec,
+    SourceFileLoader,
+)
+from inspect import isclass
+
 from coalib.collecting.Importers import import_objects
 
 
 class ImportObjectsTest(unittest.TestCase):
 
     def setUp(self):
         current_dir = os.path.split(__file__)[0]
-        self.testfile1_path = os.path.join(current_dir,
-                                           'importers_test_dir',
+        self.test_dir = os.path.join(current_dir, 'importers_test_dir')
+        self.testfile1_path = os.path.join(self.test_dir,
                                            'file_one.py')
-        self.testfile2_path = os.path.join(current_dir,
-                                           'importers_test_dir',
+        self.testfile2_path = os.path.join(self.test_dir,
                                            'file_two.py')
 
+    def check_imported_file_one_test(self, obj):
+        self.assertTrue(isclass(obj))
+
+        self.assertEqual(obj.__name__, 'test')
+        self.assertEqual(obj.__module__, 'file_one')
+
+        instance = obj()
+        self.assertIsInstance(instance, list)
+
     def test_no_file(self):
         self.assertEqual(import_objects([]), [])
 
     def test_no_data(self):
-        self.assertEqual(len(import_objects(self.testfile1_path)), 12)
+        obj = import_objects(self.testfile1_path)
+        self.assertIsInstance(obj, list)
+        self.assertEqual(len(obj), 12)
+
+        self.assertIsInstance(obj[0], dict)
+        self.assertIn('__name__', obj[0])
+        self.assertEqual(obj[0]['__name__'], 'builtins')
+        self.assertIn('copyright', obj[0])
+        self.assertIsInstance(obj[1], str)
+        self.assertTrue(obj[1].endswith('.pyc'))
+        self.assertTrue(obj[1].startswith(self.test_dir))
+        self.assertIsNone(obj[2])
+        self.assertIsInstance(obj[3], str)
+        self.assertEqual(obj[3], self.testfile1_path)
+        self.assertIsInstance(obj[4], SourceFileLoader)
+        self.assertIsInstance(obj[5], str)
+        self.assertEqual(obj[5], 'file_one')
+        self.assertIsInstance(obj[6], str)
+        self.assertEqual(obj[6], '')
+        self.assertIsInstance(obj[7], ModuleSpec)
+        self.assertIsInstance(obj[8], list)
+        self.assertEqual(obj[8], [1, 2, 3])
+        self.assertIsInstance(obj[9], list)
+        self.assertEqual(obj[9], [1, 2, 4])
+        self.assertIsInstance(obj[10], bool)
+        self.assertIs(obj[10], True)
+        self.check_imported_file_one_test(obj[11])
```

Comment:
Would be nice to have some comments showing the member names behind the list indices above. Or maybe even `is` relations by using the actual module object for verification:

```python
objs = import_objects(self.testfile1_path)
module = _import_module(self.testfile1_path)

assertIs(objs[0], module.__dict__)
...
```

That would also implicitly check that every module `file_path` is really imported only once

Applies to other `test_` methods as well

Reply:
Using `_import_module` to check `import_objects` would defeat the purpose of the test.

Reply:
@jayvdb So let's keep it like this. Especially since we need some further discussion about the `_import_module` behavior at all according to https://github.com/coala/coala/pull/4210#issuecomment-344947186

cc @Makman2